src/cache-utils.ts

```diff
@@ -66,35 +104,123 @@ export const getPackageManagerInfo = async (packageManager: string) => {
   } else if (packageManager === 'pnpm') {
     return supportedPackageManagers.pnpm;
   } else if (packageManager === 'yarn') {
-    const yarnVersion = await getPackageManagerVersion('yarn', '--version');
-
-    core.debug(`Consumed yarn version is ${yarnVersion}`);
-
-    if (yarnVersion.startsWith('1.')) {
-      return supportedPackageManagers.yarn1;
-    } else {
-      return supportedPackageManagers.yarn2;
-    }
+    return supportedPackageManagers.yarn;
   } else {
     return null;
   }
 };
 
-export const getCacheDirectoryPath = async (
+/**
+ * glob expanding memoized because it involves potentially very deep
+ * traversing through the directories tree
+ */
+export const memoizedCacheDependencies: Record<string, string[]> = {};
+/**
+ * Expands (converts) the string input `cache-dependency-path` to list of directories that
+ * may be project roots
+ * @param cacheDependencyPath - either a single string or multiline string with possible glob patterns
+ *                              expected to be the result of `core.getInput('cache-dependency-path')`
+ * @return list of directories and possible
+ */
+const getProjectDirectoriesFromCacheDependencyPath = async (
+  cacheDependencyPath: string
+): Promise<string[]> => {
+  let cacheDependenciesPaths: string[];
+
+  // memoize unglobbed paths to avoid traversing FS
+  const memoized = memoizedCacheDependencies[cacheDependencyPath];
+  if (memoized) {
+    cacheDependenciesPaths = memoized;
+  } else {
+    const globber = await glob.create(cacheDependencyPath);
+    cacheDependenciesPaths = (await globber.glob()) || [''];
+    memoizedCacheDependencies[cacheDependencyPath] = cacheDependenciesPaths;
+  }
+
+  const existingDirectories: string[] = cacheDependenciesPaths
+    .map(path.dirname)
+    .filter(path => path != null)
+    .filter(unique())
+    .filter(fs.existsSync)
+    .filter(directory => fs.lstatSync(directory).isDirectory());
+
+  // if user explicitly pointed out some file, but it does not exist it is definitely
+  // not he wanted, thus we should throw an error not trying to workaround with unexpected
+  // result to the whole build
+  if (existingDirectories.length === 0)
+    throw Error(
```

Comment:
Could you please change it to warning.

Reply:
Besides the motivation in the comment explaining why there should be an error the changing it to waring implies returning empty array that will cause the vague error anyway https://github.com/actions/toolkit/blob/main/packages/cache/src/cache.ts#L27

I need one more confirmation throwing the error should be replaced with warning